### DISCUSSION TITLE
FileDrop widget Drag and Drop files into Textual apps

### DISCUSSION THREAD

**Discussion by agmmnn** [Textualize/textual#1414] [Show and tell]
I created a widget that can get file directories through drag and drop using the [Paste ](https://textual.textualize.io/events/paste/) event. I only tested it in the `Windows Terminal`, so I don't know if it works in other terminals/operating systems. https://github.com/microsoft/terminal/pull/4323

https://github.com/agmmnn/textual-filedrop

![](https://user-images.githubusercontent.com/16024979/208706132-0a33bb21-51b8-441a-aeb9-668dbfcb382c.gif)

**Comment by willmcgugan** [comment]
Nice. I didn't know this was possible. We will try it on Mac and Linux.

**Comment by davep** [comment]
Works nicely on macOS:

<img width="922" alt="Screenshot 2022-12-21 at 11 55 41" src="https://user-images.githubusercontent.com/28237/208899650-d2b55c7a-9951-4282-8c50-40cb6af51616.png">

@agmmnn -- when I first tried the example I got:

```
No module named 'tldextract'
```

Fixed it easy enough with a:

```sh
$ poetry add tldextract
```

Perhaps `tldextract` should be some sort of dev dependency or something?

**Comment by agmmnn** [comment]
thank you @davep i added `tldextract` as dev dependency and macOS test info to readme.
